### PR TITLE
Removing Sync Halibut from Tentacle 

### PR DESCRIPTION
**Pull request by sburmanoctopus**
# Background

Sync operations have been removed from Halibut in favour of Async operations. 

This PR updates the version of Halibut used in the Tentacle solution to one with Sync Halibut removed.

This PR also changes Tentacle to support only Async Halibut and updates the Services implemented in Tentacle to use the Async Service Contract along with modifications to the implementation to use async methods where it makes sense.

# Results

The Tentacle solution has all Sync Halibut operations removed.

Related to OctopusDeploy/Issues#8266

[sc-58528]
[sc-56576]

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered the [appropriate target version for this PR](https://octopushq.atlassian.net/wiki/spaces/RND/pages/1145896982/Which+versions+should+I+patch#Process).
- [ ] I have considered appropriate testing for my change.
  - Automated testing / Exploratory testing / Nothing required?
  - Is the build green?
- [ ] I have considered manually testing my changes on a [branch instance](https://octopushq.atlassian.net/wiki/spaces/RND/pages/1262256152/Use+a+Core+Feature+Branch+Instance) to check correctness, stability and performance on Octopus Cloud.
- [ ] I have considered safety nets to reduce any time-to-recovery for my change.

### CODE REVIEW

File: docker/windows/Scripts/configure-tentacle.ps1 [hunk 1]
```diff
@@ -112,7 +113,11 @@ function Validate-Variables() {
   Write-Log " - server endpoint '$ServerUrl'"
   Write-Log " - api key '##########'"
   if (($null -ne $ServerCommsAddress) -or ($null -ne $ServerPort)) {
-    Write-Log " - communication mode 'Polling' (Active)"
+    if ($null -ne $AsKubernetesTentacle ) {
+      Write-Log " - communication mode 'Kubernetes' (Polling)"
```
Comment (LukeButters) [inline docker/windows/Scripts/configure-tentacle.ps1:117]: Why are we adding this?

File: docker/linux/scripts/configure-tentacle.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ mkdir -p $configurationDirectory
 mkdir -p $applicationsDirectory
```
Comment (LukeButters) [inline docker/linux/scripts/configure-tentacle.sh:17]: ugh just formatting changes :)

File: source/Octopus.Tentacle.Tests/Capabilities/CapabilitiesServiceV2Fixture.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ public class CapabilitiesServiceV2Fixture
         public void CapabilitiesAreReturned()
         {
             var capabilities = new CapabilitiesServiceV2()
```
Comment (LukeButters) [inline source/Octopus.Tentacle.Tests/Capabilities/CapabilitiesServiceV2Fixture.cs:12]: We don't need an await here?

File: source/Octopus.Tentacle/Services/Scripts/ScriptService.cs [hunk 4]
```diff
@@ -36,7 +36,8 @@ public ScriptTicket StartScript(StartScriptCommand command)
                 command.ScriptIsolationMutexTimeout,
                 command.IsolationMutexName,
                 command.Arguments,
-                command.Files);
+                command.Files,
+                CancellationToken.None).Result;
```
[the remark this answers is not in the document]
Comment (LukeButters) [reply source/Octopus.Tentacle/Services/Scripts/ScriptService.cs:51]: @sburmanoctopus are you adding this?

File: docker/windows/Scripts/configure-tentacle.ps1 (hunk 1 above)
Comment (sburmanoctopus) [reply docker/windows/Scripts/configure-tentacle.ps1:117]: I didn't... I suspect this is a rebase/merging issue.

I'll sort this out when the branch it's going into gets merged into master.

File: source/Octopus.Tentacle/Services/Scripts/ScriptService.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (sburmanoctopus) [reply source/Octopus.Tentacle/Services/Scripts/ScriptService.cs:51]: Now that we made the services async, I thought I'd try simply awaiting it...

I also used the cancellation token passed in.

@LukeButters , let me know if this isn't a good thing to do

[the remark this answers is not in the document]
Comment (LukeButters) [reply source/Octopus.Tentacle/Services/Scripts/ScriptService.cs:51]: cool

File: source/Octopus.Tentacle.Tests/Capabilities/CapabilitiesServiceV2Fixture.cs (hunk 3 above)
Comment (sburmanoctopus) [reply source/Octopus.Tentacle.Tests/Capabilities/CapabilitiesServiceV2Fixture.cs]: We do, I just hadn't finished the work off from our pairing session. It should be there now.

File: source/Octopus.Tentacle/Services/FileTransfer/FileTransferService.cs [hunk 5]
```diff
@@ -71,9 +76,7 @@ public UploadResult UploadFile(string remotePath, DataStream upload)
             fileSystem.EnsureDiskHasEnoughFreeSpace(parentDirectory, upload.Length);
 
             log.Trace("Copying uploaded data stream to: " + fullPath);
-#pragma warning disable CS0612
-            upload.Receiver().SaveTo(fullPath);
-#pragma warning restore CS0612
+            upload.Receiver().SaveToAsync(fullPath, CancellationToken.None).Wait();
```
Comment (nathanwoctopusdeploy) [inline source/Octopus.Tentacle/Services/FileTransfer/FileTransferService.cs:79]: Should this not just await?

**Review by nathanwoctopusdeploy — APPROVED** [review]
LGTM

**Comment by nathanwoctopusdeploy** [comment]
Fixes https://github.com/OctopusDeploy/OctopusTentacle/issues/670

File: source/Octopus.Tentacle/Services/FileTransfer/FileTransferService.cs (hunk 5 above)
Comment (sburmanoctopus) [reply source/Octopus.Tentacle/Services/FileTransfer/FileTransferService.cs:79]: Indeed. I fixed it in [this PR](https://github.com/OctopusDeploy/OctopusTentacle/pull/662)

**Pull request merged**